guava/src/com/google/common/math/LongMath.java

```diff
@@ -161,7 +168,7 @@ public static int log2(long x, RoundingMode mode) {
   // TODO(kevinb): remove after this warning is disabled globally
   public static int log10(long x, RoundingMode mode) {
     checkPositive("x", x);
-    int logFloor = log10Floor(x);
+    @IndexFor(value = {"powersOf10", "halfPowersOf10"}) int logFloor = log10Floor(x);
```

Comment:
Is this annotation necessary? It should be inferred from the return type of `log10Floor`.

Reply:
As a minor note, you don't need to write `value = {...}`; you can just write `{...}` as long as there is only one argument to the annotation, and Java will infer that it should be assigned to the `value` field.